lib/cylc/gui/gscan.py

```diff
@@ -826,8 +826,7 @@ def _get_warnings(self, key):
         failed_tasks = []
         for state in (TASK_STATUS_FAILED, TASK_STATUS_SUBMIT_FAILED):
             failed_tasks += self.tasks_by_state[key].get(state, [])
-        warnings = [warn for warn in failed_tasks if warn[0] > warn_time]
-        warnings.sort()
+        warnings = sorted(warn for warn in failed_tasks if warn[0] > warn_time)
```

Comment:
[It sounds like either way a list is created, and then sorted.](https://stackoverflow.com/q/4154571/1953517). (If I'm following the answers correctly...)